Source/WebCore/platform/network/cocoa/NetworkStorageSessionCocoa.mm

```diff
@@ -172,15 +172,23 @@ - (CFURLRef)_cfurl;
     END_BLOCK_OBJC_EXCEPTIONS
 }
 
-RetainPtr<NSHTTPCookieStorage> NetworkStorageSession::nsCookieStorage() const
+RetainPtr<NSHTTPCookieStorage> NetworkStorageSession::nsCookieStorageIfExists() const
 {
     ASSERT(hasProcessPrivilege(ProcessPrivilege::CanAccessRawCookies) || m_isInMemoryCookieStore);
     auto cfCookieStorage = cookieStorage();
     ASSERT(cfCookieStorage || !m_isInMemoryCookieStore);
     if (!m_isInMemoryCookieStore && (!cfCookieStorage || [NSHTTPCookieStorage sharedHTTPCookieStorage]._cookieStorage == cfCookieStorage))
         return [NSHTTPCookieStorage sharedHTTPCookieStorage];
 
-    return adoptNS([[NSHTTPCookieStorage alloc] _initWithCFHTTPCookieStorage:cfCookieStorage.get()]);
+    return nullptr;
+}
+
+RetainPtr<NSHTTPCookieStorage> NetworkStorageSession::nsCookieStorage() const
+{
+    if (RetainPtr nsCookieStorage = nsCookieStorageIfExists())
+        return nsCookieStorage;
+
+    return adoptNS([[NSHTTPCookieStorage alloc] _initWithCFHTTPCookieStorage:cookieStorage().get()]);
```

Comment:
I don't think this PR is correct. As you can see `nsCookieStorage()` creates a `NSHTTPCookieStorage` cut doesn't cache it. So we keep constructing a new one every time unless we're using the *shared* cookie store.

I don't think your PR helps?